syntax = "proto3";

package greet;

import "google/protobuf/timestamp.proto";

// A class to show you how you can greet someone.
service GreetService {
  rpc SubscribeToNotifications(SubscribeToNotificationsRequest) returns (stream SubscribeToNotificationsResponse);
}

message SubscribeToNotificationsRequest {
  string notification_type = 1;
}
message SubscribeToNotificationsResponse {
  string type = 1;
  string title = 2;
  string content = 3;
  optional bool is_blocking = 4;
}